app/model/Google/OAuthMailer.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Model\Google;
+
+use Google_Service_Gmail;
+use Google_Service_Gmail_Message;
+use Nette\Mail\IMailer;
+use Nette\Mail\Message;
+use function base64_encode;
+use function str_replace;
+
+class OAuthMailer implements IMailer
+{
+    private Google_Service_Gmail $gmailService;
+
+    public function __construct(Google_Service_Gmail $gmailService)
+    {
+        $this->gmailService = $gmailService;
+    }
+
+    public function send(Message $mail) : void
+    {
+        $gmsg = new Google_Service_Gmail_Message();
+        $gmsg->setRaw(str_replace(['+', '/', '='], ['-', '_', ''], base64_encode($mail->generateMessage())));
```

Comment:
Proč se tady pls dělá ten replace?

Reply:
našel jsem to ve více examplech, že to takle dělali, přesný účel nevím

Reply:
Tak to by chtělo zjistit. Protože tohle je dost WTF a mít kód, kterej nevím co dělá a je jen copy-paste, je IMO špatně.

Reply:
Tomu rozumim, ale tohle byl zpusob jak posilat maily pomoci setRaw. Našel jsem zdroj https://github.com/glennfriend/gmail-import-use-api/blob/master/app/components/manager/GmailManager.php#L237 Chapal jsem to jako problemove znaky, ale muzeme to odebrat.

Reply:
Jo, už to chápu. To musí být URL safe encoding.

Mohl bys to prosím vytáhnout třeba do privátní metody, ať tam pak je něco jako:
```suggestion
        $gmsg->setRaw($this->urlSafeBase64Encode($mail->generateMessage()));
```

Reply:
jasně, upraveno